### PR TITLE
EOS-17966: Implemented event manager publish method

### PR DESCRIPTION
**Pull request by indrajitzagade**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    EOS-17966: Implemented event manager publish method
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes (Tested by running the publisher test directly without building/installing rpm)
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Your Problem decription here...
  </code>
</pre>
## Solution
<pre>
  <code>
    Implement interface to publish HA events on the message bus
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    Ran the publisher test successfully.
  </code>
</pre>

### CODE REVIEW

**Comment by indrajitzagade** [comment]
To avoid duplicate efforts, waiting for PR #509 and #510 to get merged. Some update to current PR will be needed after this merge.

File: ha/core/event_manager/event_manager.py
Comment (ajay-paratmandali) [inline ha/core/event_manager/event_manager.py:151]: Can we saperate message bus and this code saperate
message bus needed many place so want to centralize it

https://github.com/Seagate/cortx-ha/pull/514/files
created class under util for same reason

File: ha/util/message_bus.py
Comment (indrajitzagade) [inline ha/util/message_bus.py:1]: This file has been copied from PR #514

File: ha/core/event_manager/event_manager.py
Comment (indrajitzagade) [reply ha/core/event_manager/event_manager.py]: done, using the class available in PR #514

**Review by ajaysrivas — APPROVED** [review]

**Review by ajay-paratmandali — APPROVED** [review]

**Comment by ajay-paratmandali** [comment]
Please resolve conflict 
@indrajitzagade

**Comment by indrajitzagade** [comment]
> 
> 
> Please resolve conflict
> @indrajitzagade

Resolved @ajaysrivas

File: ha/core/event_manager/event_manager.py
Comment (ajay-paratmandali) [inline ha/core/event_manager/event_manager.py:346]: can we have _prepare_key method to get this keys

Comment (indrajitzagade) [reply ha/core/event_manager/event_manager.py:346]: If you mean the _prepare_key() method in system health module, then I guess lets not it now.
We will have to make it generic before use.

File: ha/core/event_manager/event_manager.py
Comment (ArchanaLimaye) [inline ha/core/event_manager/event_manager.py:347]: can skip this check.
 _confstore.get will return "None" if key is not found and that "None" can be used to skip the publish call

Comment (ArchanaLimaye) [inline ha/core/event_manager/event_manager.py:347]: can skip this check.
_confstore.get will return "None" if key is not found and that "None" can be used to skip the publish call

File: ha/core/event_manager/const.py
Comment (indrajitzagade) [reply ha/core/event_manager/const.py]: Good catch, deleted

File: ha/core/event_manager/event_manager.py
Comment (indrajitzagade) [reply ha/core/event_manager/event_manager.py]: Done

**Review by ArchanaLimaye — APPROVED** [review]

**Pull request merged**
